frontend/src/components/me/config/UserConfig.jsx

```diff
@@ -0,0 +1,332 @@
+import { addToast, useAxiosComponentLoader } from "@certego/certego-ui";
+import { Field, FieldArray, Form, Formik } from "formik";
+import PropTypes from "prop-types";
+import React from "react";
+import {
+  BsFillCheckSquareFill,
+  BsFillPencilFill,
+  BsFillPlusCircleFill,
+  BsFillTrashFill,
+} from "react-icons/bs";
+import { MdCancel } from "react-icons/md";
+import { Row, Col, Container, FormGroup, Input, Button } from "reactstrap";
+import useTitle from "react-use/lib/useTitle";
+import { usePluginConfigurationStore } from "../../../stores";
+import {
+  createCustomConfig,
+  CUSTOM_CONFIG_URI,
+  deleteCustomConfig,
+  updateCustomConfig,
+} from "./api";
+
+function isJSON(str) {
+  if (str === '""') return true;
+  try {
+    return Boolean(JSON.parse(str) && !!str);
+  } catch (e) {
+    return false;
+  }
+}
+
+export function isValidConfig(item) {
+  if (!item.type) {
+    addToast("Invalid config!", "Please select a type", "danger", true);
+    return false;
+  }
+  if (!item.plugin_name) {
+    addToast("Invalid config!", "Please select a plugin", "danger", true);
+    return false;
+  }
+  if (!item.attribute) {
+    addToast("Invalid config!", "Please select an attribute", "danger", true);
+    return false;
+  }
+  if (!isJSON(item.value)) {
+    addToast(
+      "Invalid config!",
+      "Please enter a JSON-compliant value",
+      "danger",
+      true
+    );
+    return false;
+  }
+  return true;
+}
+
+export function filterEmptyParams(plugins) {
+  return Object.keys(plugins)
+    .filter(
+      (key) =>
+        plugins[key].params && Object.keys(plugins[key].params).length > 0
+    )
+    .reduce((res, key) => Object.assign(res, { [key]: plugins[key] }), {});
+}
+
+export function Config({ configFilter, additionalConfigData }) {
+  const [
+    analyzers,
+    connectors,
+    retrieveAnalyzersConfiguration,
+    retrieveConnectorsConfiguration,
+  ] = usePluginConfigurationStore((state) => [
+    filterEmptyParams(state.analyzersJSON),
+    filterEmptyParams(state.connectorsJSON),
+    state.retrieveAnalyzersConfiguration,
+    state.retrieveConnectorsConfiguration,
+  ]);
+
+  const [respData, Loader, refetchConfig] = useAxiosComponentLoader(
+    {
+      url: CUSTOM_CONFIG_URI,
+    },
+    configFilter
+  );
+
+  const refetchAll = () => {
+    refetchConfig();
+    retrieveAnalyzersConfiguration();
+    retrieveConnectorsConfiguration();
+  };
+
+  return (
+    <Loader
+      render={() => (
+        <Formik initialValues={{ config: respData }} onSubmit={null}>
+          {({ values: configurations, setFieldValue }) => (
+            <Form>
+              <FieldArray name="config">
+                {({ remove, push }) => (
+                  <FormGroup row>
+                    <Col>
+                      {configurations.config && configurations.config.length > 0
+                        ? configurations.config.map((plugin, index) => {
+                            let plugins;
```

Comment:
If we replace `plugin` with `configuration`, we'll end up having a `configurations` and a `configuration`.

Reply:
That's true, but I think is better `configuation` for thes two reasos:

* in fact `configurations` is a list and `plugin` is an element of the list, so it makes more sense to be `configuration`.
* In the other part of the code we have something like:
  ```
  if (
    plugin.plugin_name &&
    plugins[plugin.plugin_name]
  )
    attributeList = Object.keys(
      plugins[plugin.plugin_name].params
    );
  ```
  I think is harder to read because the two variables has only a "s" of difference

Reply:
@drosetti Done